Add delete user endpoint

tfnribeiro opened the pull request:
+ Expose a way to delete users to the API
+ Uses the user current session id to start the deletion process. 

Used by: https://github.com/zeeguu/web/pull/373

mircealungu commented on zeeguu/api/endpoints/accounts.py:
```diff
@@ -13,6 +13,25 @@
+    from zeeguu.core.account_management.user_account_deletion import delete_user_account
+
+    try:
+        session_id = int(session_id)
```
why do we pass the session_id in the URL? Can't we use the @with_session decorator?

tfnribeiro replied:
That's a good point -  I will check if it works using the decorator!

mircealungu reviewed (changes requested):
Tiago: one more observation on this PR

tfnribeiro commented:
I have changed to use the session. I also added a test case for the user deletion. I remember we talked about a possible issue with ordering, but I don't recall the solution? Right now, it is at the end of all the `user_tests`, where it deletes the `annonymous` user.

mircealungu commented on zeeguu/core/account_management/user_account_deletion.py:
```diff
@@ -0,0 +1,99 @@
+)
+from zeeguu.core.model import Article
+
+tables_to_modify = [
```
I remember seeing this elsewhere. Ah, yes, in `delete_dev_users`. Would it be possible to remove it from there, and thus, remove the duplication between the two files. I think also the `delete_user_account` method is duplicated with `delete_user`?

mircealungu reviewed (changes requested):
small favor to ask :)
